tfs/src/main/java/hudson/plugins/tfs/listeners/JenkinsRunListener.java

```diff
@@ -38,8 +56,24 @@ public void onFinalized(final Run run) {
     @Override
     public void onCompleted(final Run run, @Nonnull final TaskListener listener) {
         log.info("onCompleted: " + run.toString());
+
+        GitStatusState runGitState;
+        final Result runResult = run.getResult();
+        if (runResult.isBetterOrEqualTo(Result.SUCCESS)) {
+            runGitState = GitStatusState.Succeeded;
+        } else {
+            runGitState = GitStatusState.Failed;
+        }
+        Job currJob = run.getParent();
+        final String targetUrl = currJob.getAbsoluteUrl() + (currJob.getNextBuildNumber() - 1);
+        trySetPullRequestStatus(run, listener, runGitState, "Jenkins CI build completed", targetUrl);
```

Comment:
This looks like it's going to end up showing the same status even if 10 different jobs are launched.  Is that the case?

Reply:
Great suggestion. Defaults context string as job names are great solution. I'll change to make jobs kicked by one PR have there distinct status entries in PR UI, grouped by the contexts.

Reply:
Made changes to property support multi-job/one PR scenario. Screenshot is like below:
![prstatusui2](https://user-images.githubusercontent.com/17461693/30499188-5c75259a-9a0e-11e7-823e-f6b198afbbe7.jpg)

@mmitche , would you mind taking a 2nd look? Thanks.